gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -207,6 +209,160 @@ def __repr__(self):
         [str(s) for s in [self.name, self.genotype, self.phaseset, self.info]])
 
 
+class ToVcfRecordCoder(coders.Coder):
+  """Coder for encoding :class:`Variant` objects as .vcf text lines."""
+
+  def _get_variant_info(self, variant):
+    info_list = []
+    # Set END in info if it doesn't match start+len(reference_bases). This is
+    # usually the case for non-variant regions.
+    if (variant.start and variant.reference_bases and
+        variant.start + len(variant.reference_bases) < variant.end):
+      info_list.append('END=%d' % variant.end)
+    # Set all other fields of info.
+    for k, v in variant.info.iteritems():
+      if isinstance(v.data, list):
+        info_list.append('%s=%s' % (k, ','.join([str(x) for x in v.data])))
+      elif v.data is True:
+        info_list.append(k)
+      else:
+        info_list.append('%s=%s' % (k, str(v.data)))
+    return info_list
+
+  def _is_alternate_phaseset(self, call):
+    return call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE
+
+  def _get_variant_format_keys(self, variant):
+    format_keys = []
+    if variant.calls:
+      format_keys.append(GENOTYPE_FORMAT_KEY)
+      # If any calls have a set phaseset that is not `DEFAULT_PHASESET_VALUE`,
+      # the key will be added to the format field.
+      if any(self._is_alternate_phaseset(call) for call in variant.calls):
+        format_keys.append(PHASESET_FORMAT_KEY)
+
+      for call in variant.calls:
+        format_keys.extend([k for k in call.info if k not in format_keys])
+
+      sort_index = 2 if PHASESET_FORMAT_KEY in format_keys else 1
+      format_keys[sort_index:] = sorted(format_keys[sort_index:])
+
+    return format_keys
+
+  def _field_missing(self, value):
+    """Returns True for empty list, string or None and False otherwise."""
+    return not value and value != 0
+
+  def _get_variant_calls(self, variant, format_keys):
+    calls_values = []
+
+    for call in variant.calls:
+      # Set GT field.
+      call_info_fields = []
+      for key in format_keys:
+        if key == GENOTYPE_FORMAT_KEY:
+          genotype_field = MISSING_FIELD_VALUE
+          if call.genotype != [MISSING_GENOTYPE_VALUE]:
+            phase_char = '|' if call.phaseset else '/'
+            genotype_field = phase_char.join([str(x) for x in call.genotype])
```

Comment:
actually, the missing value can be inside the genotype as well (e.g. "0/.").

Reply:
Can we change this condition to `MISSING_GENOTYPE_KEY not in call.genotype` then?